### PR TITLE
[MPS] Add `MetalShaderLibrary::getFunctionNames()`

### PR DESCRIPTION
**Pull request by malfet**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #141478
* #141547
* __->__ #141499

That returns names of all the function in shader

### CODE REVIEW

**Review by manuelcandales — APPROVED** [review]

File: aten/src/ATen/native/mps/OperationUtils.mm [hunk 1]
```diff
@@ -853,6 +853,20 @@ void executeMPSAllocatorCallback(void* ptr, EventType event) override {}
   return cplMap[key];
 }
 
+std::vector<std::string> MetalShaderLibrary::getFunctionNames() {
+  if (C10_UNLIKELY(!library && nparams > 0)) {
+    throw std::runtime_error("Library must be initialized first");
+  }
+  std::vector<std::string> rc;
+  @autoreleasepool {
+    NSArray<NSString*>* names = [getLibrary() functionNames];
```
Comment (Skylion007) [inline aten/src/ATen/native/mps/OperationUtils.mm:862]:
```suggestion
    NSArray<NSString*>* names = [getLibrary() functionNames];
    rc.reserve([names count]);
```

File: aten/src/ATen/native/mps/OperationUtils.mm [hunk 2]
```diff
@@ -853,6 +853,20 @@ void executeMPSAllocatorCallback(void* ptr, EventType event) override {}
...
+    for (auto idx : c10::irange([names count])) {
+      rc.push_back([[names objectAtIndex:idx] UTF8String]);
```
Comment (Skylion007) [inline aten/src/ATen/native/mps/OperationUtils.mm:864]:
```suggestion
      rc.emplace_back([[names objectAtIndex:idx] UTF8String]);
```

**Review by Skylion007 — APPROVED** [review]
Looks good, modulo nits.

**Pull request closed without merging**